cogeo_mosaic/backend/dynamo.py

```diff
@@ -0,0 +1,88 @@
+import functools
+import itertools
+import os
+from typing import Dict, Tuple
+
+import mercantile
+
+from cogeo_mosaic.backend.base import BaseBackend
+
+
+class DynamoDBBackend(BaseBackend):
+    """DynamoDB Backend Adapter"""
+
+    def __init__(
+        self, mosaicid: str, region: str = os.getenv("AWS_REGION", "us-east-1")
+    ):
+        self.client = boto3.resource("dynamodb", region_name=region)
+        self.table = self.client.Table(mosaicid)
+        self.mosaic_def = self.fetch_mosaic_definition()
```

Comment:
maybe fecth_mosaic_definion could just feed properties define here https://github.com/developmentseed/cogeo-mosaic/pull/38/files#r398594144

Reply:
I think it's easiest to return a `dict` from `fetch_mosaic_definition`... Otherwise you'd be returning 8 values that you need to unpack into `self` right? Like
```py
self.minzoom, self.maxzoom, self.quadkey_zoom ... = self.fetch_mosaic_definition()
```
Is that what you mean?

Reply:
if you define the values in init in the base class you can then do 
```
class Base(ABC):
def __init__(self, *args, **kwargs):
    self.minzoom: int
    self.maxzoom: int
    self.quadkey_zoom: Optional[int]
    self.bounds: Tuple[float, float, float, float]
    self.center: Optional[Tuple[float, float, int]]
    sefl.version: Optional[str]
    self.name: Optional[str]
    self.mosaicjson_version: str
```
```
class backend(Base):
    def __init__(self, url, ...):
        super(backend, self).__init__()
        self._fetch_mosaic_def(url)
```

 the `_fetch_mosaic_def` will do:

```
def _fetch_mosaic_def(self, url):
    mosaic = fetch(...)
    self.minzoom = mosaic["minzoom"]
....

```

Reply:
this is maybe totally not the best way to do it (just thinking how I would have done it :D )

Reply:
Currently the only one of those we actually use is `quadkey_zoom` (derived from it and `minzoom`). So I defined only `quadkey_zoom` in `__init__` and if we need more values in the future we can add them then. Sound good?